src/gqlRules/gqlNameMatch/gqlNameMatch.js

```diff
@@ -0,0 +1,95 @@
+import { isGqlFile } from "../utils";
+import { isQuery, isMutation } from "../utils/isGQLOperation";
+import { relativePathToFile } from "../../utils";
+import { operationName } from "../utils/operationName";
+import { sanitizeGqlOperationText } from "../utils/sanitizeGqlOperationText";
+
+const meta = {
+  type: "problem",
+  docs: {
+    description:
+      "The Variable name GQL operation is assigned should match with the GQL operation name Ex: const Patient_Query_UserAvatar = gql` query Patient_Query_UserAvatar...`",
+    category: "gql-name-match",
+    recommended: false,
+  },
+  fixable: "code",
+  schema: [
+    {
+      type: "object",
+      properties: {
+        namespaceIgnoreList: {
+          type: "array",
+          items: {
+            type: "string",
+          },
+        },
+        namespaceOperationPrefix: {
+          type: "object",
+        },
+      },
+      required: ["namespaceOperationPrefix"],
+      additionalProperties: false,
+    },
+  ],
+};
+
+const create = context => {
+  const isGqlObjectFile = isGqlFile(context);
+  const { namespaceOperationPrefix, namespaceIgnoreList } = context.options[0];
+
+  const isGqlTemplateElement = node => {
+    return node.tag && node.tag.name === "gql";
+  };
+
+  const getNamespace = () => {
+    const pathToFile = relativePathToFile(context);
+    const relativeJsPath = pathToFile.replace("app/javascript/", "");
+
+    const namespace = Object.keys(namespaceOperationPrefix).find(namespaceOperationKey =>
+      relativeJsPath.startsWith(namespaceOperationKey)
+    );
+    return { namespace };
+  };
+
+  const isInIgnoreList = namespace => {
+    return namespaceIgnoreList.includes(namespace);
+  };
+
+  const validateOperationName = (gqlOperationText, operationType, node) => {
```

Comment:
I feel like based on all my comments, we should rename this function as it's not as descriptive as to what it's doing anymore. 

Perhaps something like: `IsOperationNameAndVariableNameSame` or something like that?